src/z3c/sqlalchemy/base.py

```diff
@@ -137,5 +136,4 @@ def _createEngine(self):
                                             autocommit=not self.transactional,
                                             twophase=self.twophase,
                                             autoflush=True,
-                                            extension=ZopeTransactionExtension(**self.extension_options),
```

Comment:
This is unrelated to the Python 3 port and should be a part of a separate PR.

Reply:
Got it, I just removed it because it was failing, anyway here's the path
https://github.com/zopefoundation/zope.sqlalchemy/blob/master/CHANGES.rst#12-2019-10-17

Reply:
It is not enough to remove ZopeTransactionExtension, you also have to call zope.sqlalchemy.register().

A separate PR that does just that would be nice.